flang/lib/Lower/Bridge.cpp

```diff
@@ -205,6 +205,19 @@ class FirConverter : public Fortran::lower::AbstractConverter {
       declareFunction(f);
   }
 
+  /// Return the host symbol if this is the symbol of an internal procedure
+  /// variable that belongs to the host procedure.
+  const Fortran::semantics::Symbol *
+  getIfHostEscapingSymbol(const Fortran::semantics::Symbol &sym) {
```

Comment:
Do we need to be precise that this is a _host procedure_ here? (Distinguish it from a host module?)

Reply:
Right, I went for `getIfHostProcedureSymbol` and actually modified the code because as you pointed out, "belongs to module" does not imply "use associated", so my logic was broken if the front-end ever starts using HostAssociated symbols for module variables inside contained procedures.